pac4j-oauth/src/main/java/org/pac4j/oauth/profile/linkedin2/LinkedIn2Profile.java

```diff
@@ -1,119 +1,105 @@
 
 /**
- * <p>This class is the user profile for LinkedIn with appropriate getters.</p>
- * <p>It is returned by the {@link org.pac4j.oauth.client.LinkedIn2Client}.</p>
  *
- * @author Jerome Leleu
- * @since 1.4.1
+ * @author Vassilis Virvilis
  */
-@SuppressWarnings("unchecked")
 public class LinkedIn2Profile extends OAuth20Profile {
-    
-    private static final long serialVersionUID = -2652388591255880018L;
-    
-    public String getOAuth10Id() {
-        String url = getSiteStandardProfileRequest();
-        return CommonHelper.substringBetween(url, "id=", "&authType=");
+    private static final long serialVersionUID = 100L;
+
+    public String getLocalizedFirstName() {
+        return (String) getAttribute(LinkedIn2ProfileDefinition.LOCALIZED_FIRST_NAME);
+    }
+
+    public String getLocalizedLastName() {
+        return (String) getAttribute(LinkedIn2ProfileDefinition.LOCALIZED_LAST_NAME);
+    }
+
+    public LinkedIn2ProfilePicture getProfilePicture() {
+        return (LinkedIn2ProfilePicture) getAttribute(LinkedIn2ProfileDefinition.PROFILE_PICTURE);
+    }
+
+    public LinkedIn2ProfileEmails getProfileEmails() {
+        return (LinkedIn2ProfileEmails) getAttribute(LinkedIn2ProfileDefinition.PROFILE_EMAILS);
     }
 
     @Override
     public String getFirstName() {
-        return (String) getAttribute(LinkedIn2ProfileDefinition.FIRST_NAME);
+        return getLocalizedFirstName();
     }
-    
+
     @Override
     public String getFamilyName() {
-        return (String) getAttribute(LinkedIn2ProfileDefinition.LAST_NAME);
+        return getLocalizedLastName();
     }
-    
+
     @Override
     public String getDisplayName() {
-        return (String) getAttribute(LinkedIn2ProfileDefinition.FORMATTED_NAME);
-    }
-    
-    @Override
-    public String getLocation() {
-        LinkedIn2Location location = (LinkedIn2Location) getAttribute(LinkedIn2ProfileDefinition.LOCATION);
-        if (location != null) {
-            return location.getName();
-        } else {
-            return null;
-        }
+        return getFirstName() + " " + getFamilyName();
     }
-    
-    @Override
-    public String getEmail() {
-        return (String) getAttribute(LinkedIn2ProfileDefinition.EMAIL_ADDRESS);
-    }
-    
+
     @Override
     public URI getPictureUrl() {
-        return (URI) getAttribute(LinkedIn2ProfileDefinition.PICTURE_URL);
-    }
-    
-    @Override
-    public URI getProfileUrl() {
-        return (URI) getAttribute(LinkedIn2ProfileDefinition.PUBLIC_PROFILE_URL);
-    }
-    
-    public LinkedIn2Location getCompleteLocation() {
-        return (LinkedIn2Location) getAttribute(LinkedIn2ProfileDefinition.LOCATION);
-    }
-    
-    public String getMaidenName() {
-        return (String) getAttribute(LinkedIn2ProfileDefinition.MAIDEN_NAME);
-    }
+        final LinkedIn2ProfilePicture pp = getProfilePicture();
+        if (pp == null)
```

Comment:
It's written nowhere, but I'd prefer to have curly brackets even for single line.